### PR TITLE
Skype: remove old detection logic

### PR DESCRIPTION
**Pull request by IvanNardi**
Skype has been using standard protocols (STUN/ICE or TLS) for a long, long time, now. Long gone are the days of Skype as a distribuited protocol.

See: #2166

### CODE REVIEW

**Comment by utoni** [comment]
> Skype has been using standard protocols (STUN/ICE or TLS) for a long, long time, now. Long gone are the days of Skype as a distribuited protocol.

Is there any information available that supports your thesis? I am not familiar with Skype/Teams, but I am using those on a daily base and would be very sad if `nDPId` does not classify those flows anymore.

File: tests/cfgs/default/result/zoom.pcap.out [hunk 1]
```diff
@@ -43,7 +43,7 @@ JA3 Host Stats:
 	1	 192.168.1.117            	 4      
 
 
-	1	UDP 192.168.1.117:58327 <-> 109.94.160.99:8801 [proto: 189/Zoom][IP: 0/Unknown][Encrypted][Confidence: DPI][DPI packets: 1][cat: Video/26][10 pkts/7806 bytes <-> 175 pkts/184434 bytes][Goodput ratio: 95/96][1.44 sec][bytes ratio: -0.919 (Download)][IAT c2s/s2c min/avg/max/stddev: 0/0 14/8 32/35 11/5][Pkt Len c2s/s2c min/avg/max/stddev: 55/60 781/1054 1071/1071 444/129][PLAIN TEXT (replace)][Plen Bins: 1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,97,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
```
Comment (lucaderi) [inline tests/cfgs/default/result/zoom.pcap.out:52]: I don't think Zoom is cleartext

File: src/lib/protocols/skype.c [hunk 2]
```diff
@@ -1,128 +0,0 @@
...
-  if(flow->host_server_name[0] != '\0')
-    return;
-  
-  if(packet->udp != NULL) {    
-
-    if(flow->packet_counter < 5) {
-      u_int16_t sport = ntohs(packet->udp->source);
-      u_int16_t dport = ntohs(packet->udp->dest);
-
-      /* skype-to-skype */
```
Comment (lucaderi) [inline src/lib/protocols/skype.c]: is skype-to-skype protocol still recognized without this code? Did you check that?

File: tests/cfgs/default/result/zoom.pcap.out (hunk 1 above)
Comment (IvanNardi) [reply tests/cfgs/default/result/zoom.pcap.out:52]: It is not. Somehow while "Zoom" is correctly associated to "Encrypted", "RTP.Zoom" is associated to "ClearText", which seems wrong to me. Probably an unrelated bug; I'll try to fix it in a new version of this PR

File: src/lib/protocols/skype.c (hunk 2 above)
Comment (IvanNardi) [reply src/lib/protocols/skype.c]: If with "skype-to-skype" you mean an interaction between "Skype-app" or "Skype-from-a-broswer" to/from "Skype-app" or "Skype-from-a-broswer", yes, I checked that. Always standard STUN or TLS stuff

**Comment by IvanNardi** [comment]
> > Skype has been using standard protocols (STUN/ICE or TLS) for a long, long time, now. Long gone are the days of Skype as a distribuited protocol.
> 
> Is there any information available that supports your thesis? I am not familiar with Skype/Teams, but I am using those on a daily base and would be very sad if `nDPId` does not classify those flows anymore.

I check Skype traffic regularly on some common environments (win/android/linux with app or via browser) and I have seen only STUN/TLS for a few years.
However, I am not able to find any public statement/information about that.

Anyway, we can keep this code around a little more; no specific harm in that

File: tests/cfgs/default/result/zoom.pcap.out (hunk 1 above)
Comment (IvanNardi) [reply tests/cfgs/default/result/zoom.pcap.out:52]: @lucaderi, what do you think of https://github.com/ntop/nDPI/pull/1955?

**Comment by 0xA50C1A1** [comment]
Will there be any updates on this PR? If you need more pcap samples to confirm that Skype is only using TLS and STUN these days, I can capture some.

**Comment by utoni** [comment]
Is the old detection logic not used anymore (for sure)?
Are there any new PCAPs that support your statement?

**Comment by 0xA50C1A1** [comment]
> Is the old detection logic not used anymore (for sure)? Are there any new PCAPs that support your statement?

Yeah, here's Skype for Android traffic (login and test call).
[Skype_for_Android.zip](https://github.com/ntop/nDPI/files/14239620/Skype_for_Android.zip)

**Comment by 0xA50C1A1** [comment]
> > Is the old detection logic not used anymore (for sure)? Are there any new PCAPs that support your statement?
> 
> Yeah, here's Skype for Android traffic (login and test call). [Skype_for_Android.zip](https://github.com/ntop/nDPI/files/14239620/Skype_for_Android.zip)

I'd also capture Skype for iOS traffic as well, but I dunno how to do it.

**Comment by utoni** [comment]
What about skype for desktops?

**Comment by IvanNardi** [comment]
I tested win11 (both via browser and via app) and ubuntu (browser) to from android/landline: standard STUN stuff.
I didn't check any Apple device/OS, though.

Examples:
[skype_win11_app_8.96.0.409_call_to_lte.zip](https://github.com/ntop/nDPI/files/14239811/skype_win11_app_8.96.0.409_call_to_lte.zip)

**Comment by 0xA50C1A1** [comment]
> What about skype for desktops?

Here's a skype for linux pcap sample

[skype_for_linux.zip](https://github.com/ntop/nDPI/files/14239764/skype_for_linux.zip)

**Review by utoni — APPROVED** [review]
Ty for all the information @0xA50C1A1 and @IvanNardi!

**Comment by IvanNardi** [comment]
Thanks everyone!

**Pull request merged**